### ISSUE TITLE
support description fields in shipment

### ISSUE THREAD

**Issue by metasnw** [metasfresh/metasfresh#1018]
### What is the current behavior?
we got the fields m_inout.description and descriptionbottom in the database but we neither show them in java client or webui in window "Lieferung".

On the shipment jasper only the descriptionbottom is displayed but not the description.

#### Which are the steps to reproduce?
create an inout
use this sql to put data
```
update m_inout 
set description = 'above', descriptionbottom = 'below'
where documentno = ???

```
### What is the expected or desired behavior?
1. both fields description and descriptionbottom should be editable in both java and webui client 
2. on jasper: description should be displayed above the inout lines like it sales order
3. on jasper: descriptionbottom should be displayed below the inout lines like it sales order (that is already working)

**Comment by metasnw** [comment]
made fields visible both in java client and webui. looks like this:
![image](https://cloud.githubusercontent.com/assets/16079655/23209279/e8499c66-f8f8-11e6-9c35-3bddd2bf07f3.png)

![image](https://cloud.githubusercontent.com/assets/16079655/23209358/45e03e48-f8f9-11e6-9e51-86da8b15034c.png)



here is the sql:

5457030_migration_script_201702221202_584967562642089344_postgresql.sql

```
-- 22.02.2017 12:02
-- URL zum Konzept
UPDATE AD_Field SET AD_FieldGroup_ID=1000002,Updated=TO_TIMESTAMP('2017-02-22 12:02:39','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=2703
;

-- 22.02.2017 12:02
-- URL zum Konzept
UPDATE AD_Field SET AD_FieldGroup_ID=1000002, IsDisplayed='Y',Updated=TO_TIMESTAMP('2017-02-22 12:02:45','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=501685
;

-- 22.02.2017 12:03
-- URL zum Konzept
UPDATE AD_Field SET IsDisplayed='Y', SeqNo=500,Updated=TO_TIMESTAMP('2017-02-22 12:03:02','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=2703
;

-- 22.02.2017 12:03
-- URL zum Konzept
UPDATE AD_Field SET SeqNo=510,Updated=TO_TIMESTAMP('2017-02-22 12:03:10','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=501685
;

-- 22.02.2017 12:05
-- URL zum Konzept
INSERT INTO AD_UI_ElementGroup (AD_Client_ID,AD_Org_ID,AD_UI_Column_ID,AD_UI_ElementGroup_ID,Created,CreatedBy,IsActive,Name,SeqNo,UIStyle,Updated,UpdatedBy) VALUES (0,0,1000014,540128,TO_TIMESTAMP('2017-02-22 12:05:22','YYYY-MM-DD HH24:MI:SS'),100,'Y','Text',30,'primary',TO_TIMESTAMP('2017-02-22 12:05:22','YYYY-MM-DD HH24:MI:SS'),100)
;

-- 22.02.2017 12:05
-- URL zum Konzept
INSERT INTO AD_UI_Element (AD_Client_ID,AD_Field_ID,AD_Org_ID,AD_Tab_ID,AD_UI_ElementGroup_ID,AD_UI_Element_ID,Created,CreatedBy,IsActive,IsAdvancedField,IsDisplayed,IsDisplayedGrid,IsDisplayed_SideList,Name,SeqNo,SeqNoGrid,SeqNo_SideList,Updated,UpdatedBy) VALUES (0,2703,0,257,540128,541803,TO_TIMESTAMP('2017-02-22 12:05:46','YYYY-MM-DD HH24:MI:SS'),100,'Y','N','Y','N','N','description',10,0,0,TO_TIMESTAMP('2017-02-22 12:05:46','YYYY-MM-DD HH24:MI:SS'),100)
;

-- 22.02.2017 12:06
-- URL zum Konzept
INSERT INTO AD_UI_Element (AD_Client_ID,AD_Field_ID,AD_Org_ID,AD_Tab_ID,AD_UI_ElementGroup_ID,AD_UI_Element_ID,Created,CreatedBy,IsActive,IsAdvancedField,IsDisplayed,IsDisplayedGrid,IsDisplayed_SideList,Name,SeqNo,SeqNoGrid,SeqNo_SideList,Updated,UpdatedBy) VALUES (0,501685,0,257,540128,541804,TO_TIMESTAMP('2017-02-22 12:06:02','YYYY-MM-DD HH24:MI:SS'),100,'Y','N','Y','N','N','description bottom',20,0,0,TO_TIMESTAMP('2017-02-22 12:06:02','YYYY-MM-DD HH24:MI:SS'),100)
;



```

**Comment by metas-ts** [comment]
..also cherry-picked it into the `release-2017-08` branch

**Comment by metasnw** [comment]
IT result
Working fine with one line text from webui and Java client

NOK: line breaks get lost in java client after logging in again

**Comment by metasnw** [comment]
fix:
switching columns from text long to text:

```

-- 24.02.2017 17:11
-- URL zum Konzept
UPDATE AD_Column SET AD_Reference_ID=14, FieldLength=1024,Updated=TO_TIMESTAMP('2017-02-24 17:11:37','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=3515
;

-- 24.02.2017 17:11
-- URL zum Konzept
INSERT INTO t_alter_column values('m_inout','Description','VARCHAR(1024)',null,'NULL')
;

-- 24.02.2017 17:11
-- URL zum Konzept
UPDATE AD_Column SET AD_Reference_ID=14, FieldLength=1024,Updated=TO_TIMESTAMP('2017-02-24 17:11:52','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=501682
;

-- 24.02.2017 17:11
-- URL zum Konzept
INSERT INTO t_alter_column values('m_inout','DescriptionBottom','VARCHAR(1024)',null,'NULL')
;


```

**Comment by metasnw** [comment]
IT NR. 2
OK: line breaks get lost in java client after logging in again
test on webui and java client are fine

known issue: text variables and text snippets do not work

**Comment by metas-rc** [comment]
Results of IT3

Tested on swing, on both mf15 and a specific client. 

On mf15 the descriptions were preserved and displayed in the report correctly, with all the new and blank lines. They were preserved also after relogging.

On the specific customer client the descriptions are not available to be set via swing and if I set them in the database they were still not displayed in the window or in the report.

Also tested on https://w101.metasfresh.com:8443. Descriptions are preserved correctly, displayed in the report, and also preserved after relogging.

OK
